theseus/optimizer/sparse_linearization.py

```diff
@@ -86,12 +86,18 @@ def __init__(
         # batched data
         self.A_val: torch.Tensor = None
         self.b: torch.Tensor = None
+
         # computed lazily by self._atb_impl() and reset to None by
         # self._linearize_jacobian_impl()
         self._Atb: torch.Tensor = None
 
+        # computed lazily by self.diagonal_scaling() and reset to None by
+        # self._linearize_jacobian_impl()
+        self._AtA_diag: torch.Tensor = None
+
     def _linearize_jacobian_impl(self):
         self._Atb = None
+        self._AtA = None
```

Comment:
`self._AtA_diag = None` here? If I remember correctly, `SparseLinearization` will be reused for the same optimization problem and we need to reinitialize `self._AtA_diag` every time.

Reply:
Oh sorry, that's a typo. Good catch!

Reply:
Fixed.